fn/result.go

```diff
@@ -0,0 +1,156 @@
+package fn
+
+import (
+	"fmt"
+	"testing"
+)
+
+// Result represents a value that can either be a success (T) or an error.
+type Result[T any] struct {
```

Comment:
This file should already be there if you use `fn v1.0.9`.

Reply:
I guess it makes sense to merge `master` into `0-19-staging` again to get this file. But would also need a replace directive in go.mod to make these changes visible to the PR.

Reply:
Ended up adding a few other items here as well. I think will spin those off into a PR into master, we merge that, then tag this new version.